build.gradle

```diff
@@ -98,22 +99,45 @@ configure(javaProjects) {
             dependency("org.opensaml:opensaml:2.6.4")
             dependency("org.python:jython-standalone:2.7.1")
             dependency("org.springframework.security.extensions:spring-security-saml2-core:1.0.2.RELEASE")
+            dependencySet(group: "com.google.protobuf", version: "3.4.0") {
+              entry "protobuf-java"
+              entry "protoc"
+            }
+            dependencySet(group: "io.grpc", version: "1.6.1") {
+              entry "grpc-core"
+              entry "grpc-netty"
+              entry "grpc-stub"
+              entry "grpc-protobuf"
+              entry "protoc-gen-grpc-java"
+              entry "grpc-testing"
+            }
+            dependencySet(group: "net.devh", version: "1.3.0-RELEASE") {
+              entry "grpc-server-spring-boot-autoconfigure"
+              entry "grpc-server-spring-boot-starter"
+            }
         }
     }
 
     ext {
         generatedSourcesDir = file("${buildDir}/generated-sources/main")
         generatedSourcesJavaDir = file(new File(generatedSourcesDir, "/java"))
+        generatedSourcesGRPCDir = file("${buildDir}/generated-grpc/main/java")
     }
 
     sourceSets {
         main {
             java {
                 srcDir generatedSourcesJavaDir
+                srcDir generatedSourcesGRPCDir
             }
         }
     }
 
+    idea.module {
```

Comment:
I don't see why this is needed? Are you depending on gradle for all your IDEA stuff? I haven't seen any issues?

Reply:
IDEA will re-generate those models and then conflict on duplicate classes.
This config only affect the IDEA project generation gradle task (i.e. importing a gradle project the way you do it is not using this snippet)